### PR TITLE
Add Python supported version, limitations doc

### PR DESCRIPTION
**Pull request by Francis-Liu**
Signed-off-by: Francis Liu <feliu@microsoft.com>

### CODE REVIEW

**Comment by bodzhang** [comment]
os.memfd_create() is not currently supported either

**Comment by bodzhang** [comment]
[All flavors of os.spawn() ](https://docs.python.org/3/library/os.html#os.spawnl) are not supported. Please note os.posix_spawn() is supported.

File: doc/PythonSupport.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Mystikos Support and Limitations for Python
+
+## Verified/Supported python versions
+As stated in our [Getting started with Python](user-getting-started-docker-python.md) document, we recommend building your containerized python application using python 3.8.11 and 3.9.7. Note that if possible, try pin your python to a specific version like 3.8.11 instead of 3.8.
+
+### Install position independent python executable
+If you run `apt install python3.8` when building a Ubuntu-based docker image, the installed python3.8 (or python3.9) executable is not a position independent executable (PIE). `myst exec <cmd>` requires the cmd to be a PIE. To overcome this, either install python using miniconda (reference our [python_webserver](../solutions/python_webserver/Dockerfile) example), or use a Debian base image instead. For example,
+```Dockerfile
+FROM python:3.9-slim
```
Comment (Francis-Liu) [inline doc/PythonSupport.md:9]: Should be 3.9.7 or something.

Comment (Francis-Liu) [reply doc/PythonSupport.md:9]: Maybe also state that we plan to support newer version.

File: doc/PythonSupport.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+### Third-party library
+In generally, choose the latest versions of third-party libraries. As we usually test Mystikos against the latest versions.
+
+## Known Limitations
+### Standard library
+To get a rough idea of whether your Python application can be run in Mystikos, you can take a look at our recorded CPython test failures on [v3.8.11](../tests/cpython-tests/test_config_v3.8.11/tests.failed) and [v3.9.7](../tests/cpython-tests/test_config_v3.9.7/tests.failed). Here are several noteworthy modules we either won't support or haven't implemented yet:
+- `os`
+  - `os.fork` - fork is not supported in Mystikos
+  - `os.spawn*` - please note that `os.posix_spawn IS supported
```
Comment (Francis-Liu) [inline doc/PythonSupport.md:20]: missing a `

File: doc/PythonSupport.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  - `os.exec*`
+  - `os.system`
+  - `os.openpty` - `/dev/pty` is not supported in Mytikos (yet)
```
Comment (Francis-Liu) [inline doc/PythonSupport.md:23]: We do have the pty support, but not passing the test case yet.

File: doc/PythonSupport.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  - `os.memfd_create`
+- `multiprocessing`
```
Comment (Francis-Liu) [inline doc/PythonSupport.md:25]: Not mention this.

**Review by Francis-Liu — COMMENTED** [review]
Put a note that this doc is not final. Contents will change after we support more features.

File: doc/PythonSupport.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
+# Mystikos Support and Limitations for Python
+**NOTE**: this is a living document that is continually updated.
+
+## Verified/Supported python versions
+As stated in our [Getting started with Python](user-getting-started-docker-python.md) document, we recommend building your containerized python application using python 3.8.11 and 3.9.7. Note that if possible, try pin your python to a specific version like 3.8.11 instead of 3.8. Even when this document was written, there are newer python release than 3.8.11 or 3.9.7. We will do our best to stablize and support mystikos on newer versions.
```
Comment (jxyang) [inline doc/PythonSupport.md:5]: The get started doc mentions 3.8 and 3.9. Are you suggesting that 3.8.11 is more stable than 3.8?

File: doc/PythonSupport.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+### Install position independent python executable
+If you run `apt install python3.8` when building a Ubuntu-based docker image, the installed python3.8 (or python3.9) executable is not a position independent executable (PIE). `myst exec <cmd>` requires the cmd to be a PIE. To overcome this, either install python using miniconda (reference our [python_webserver](../solutions/python_webserver/Dockerfile) example), or use a Debian base image instead. For example,
```
Comment (jxyang) [inline doc/PythonSupport.md:8]: Please break long longs into shorter ones (~80 characters per line).

The 3rd option is to build cpython from source following instructions at https://github.com/python/cpython#build-instructions

File: doc/PythonSupport.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+```Dockerfile
+FROM python:3.9.7-slim
+```
+
+### Third-party library
```
Comment (jxyang) [inline doc/PythonSupport.md:13]: Third-party library -> Third-party Python libraries

File: doc/PythonSupport.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+In generally, choose the latest versions of third-party libraries. As we usually test Mystikos against the latest versions.
+
+## Known Limitations
+### Standard library
```
Comment (jxyang) [inline doc/PythonSupport.md:17]: This section should be moved before the 3rd-party section. And in there, we should mention any 3rd-party Python library that calls into the unsupported standard Python APIs won't be supported either.

File: doc/PythonSupport.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+To get a rough idea of whether your Python application can be run in Mystikos, you can take a look at our recorded CPython test failures on [v3.8.11](../tests/cpython-tests/test_config_v3.8.11/tests.failed) and [v3.9.7](../tests/cpython-tests/test_config_v3.9.7/tests.failed). Here are several noteworthy modules we either won't support or haven't supported yet:
```
Comment (jxyang) [inline doc/PythonSupport.md:18]: Get rid of `won't support or`.

File: doc/PythonSupport.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+### Flask
+| Supported | Known Limitations & Caveats | Sample/Tests |
+| :---: | :--- | :--- |
+| Yes | In nginx.conf, set master_process off to avoid using syscall SYS_rt_sigsuspend() which is not hanlded by Mystikos yet<br>`master_process off;`<br>The following pytest would fail for now because of issue #503.<br>`test_instance_config.py::test_egg_installed_paths` | [solutions/python_flask_tests](https://github.com/deislabs/mystikos/tree/main/solutions/python_flask_tests) |
+
+### Azure SDK for Python
+| Supported | Packages that | Sample/Tests |
+| :---: | :--- | :--- |
+| Yes | The following packages have been verified:<br> - azure-keyvault-administration<br> - azure-keyvault-certificates<br> - azure-keyvault-keys<br> - azure-keyvault-secrets<br> - azure-mgmt-keyvault<br> - azure-identity<br> - azure-storage-file-datalake<br> - azure-storage-file-share<br> - azure-storage-queue<br> - azure-mgmt-storage<br> - azure-mgmt-storagesync<br> - azure-storage-blob | [solutions/python_azure_sdk](https://github.com/deislabs/mystikos/tree/main/solutions/python_azure_sdk) |
```
Comment (jxyang) [inline doc/PythonSupport.md:64]: Please also incorporate findings from https://github.com/deislabs/mystikos/issues/370

File: doc/PythonSupport.md (hunk 5 above)
Comment (Francis-Liu) [reply doc/PythonSupport.md:5]: Unlike tag 3.8.11, tag 3.8 always "points to" the latest version. For example, when 3.8.12 comes out, 3.8 is updated to that version. Remember we had issues when the base image updates, and customer rebuilt the container, which caused failures.

File: doc/PythonSupport.md (hunk 10 above)
Comment (Francis-Liu) [reply doc/PythonSupport.md:86]: Done.

File: doc/PythonSupport.md (hunk 9 above)
Comment (Francis-Liu) [reply doc/PythonSupport.md]: Done.

File: doc/PythonSupport.md (hunk 7 above)
Comment (Francis-Liu) [reply doc/PythonSupport.md]: Fixed.

File: doc/PythonSupport.md (hunk 6 above)
Comment (Francis-Liu) [reply doc/PythonSupport.md]: Done.

File: doc/PythonSupport.md (hunk 8 above)
Comment (Francis-Liu) [reply doc/PythonSupport.md]: Done.

File: doc/PythonSupport.md [hunk 11]
```diff
@@ -31,34 +45,58 @@ To get a rough idea of whether your Python application can be run in Mystikos, y
...
+### 2.7. Other popular Python packages
+We haven't verified the full test suites of the following packages. Their support status are based on our experience of running applications that call these packages.
+| Packages | Supported |
+| :--- | :--- |
+| **logzero** | Yes |
+| **pycrypto** | Yes |
+| **pyjwt** | Yes |
+| **nose** | Yes |
+| **redis** | Yes |
+| **pyodbc** | only works with SQL AE |
```
Comment (jxyang) [inline doc/PythonSupport.md:97]: **pyodbc** should work with any Data base server. We verified it works with SQL Server, with or without the Always encrypted feature.

**Review by jxyang — APPROVED** [review]

**Pull request merged**